File: py/1.twoSum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hash = dict()
        n = len(nums)
        for i in range(n):
            if target-nums[i] in hash:
                return [hash[target-nums[i]],i]
            hash[nums[i]] = i   # 注意细节 不是hash[target - nums[i]] = i
        return []

File: golang/1.twoSum.go
func twoSum(nums []int, target int) []int {
	hashTable := map[int]int{}
	for i,value := range nums {
		if p,ok := hashTable[target-value];ok {
			return []int{p,i}
		}else {
			hashTable[value] = i
		}
	}
	return []int{}
}